apps/prairielearn/src/middlewares/authzWorkspaceCookieSet.js

```diff
@@ -17,7 +18,7 @@ module.exports = (req, res, next) => {
   res.cookie(cookieName, cookieData, {
     maxAge: config.workspaceAuthzCookieMaxAgeMilliseconds,
     httpOnly: true,
-    secure: true,
+    secure: shouldSecureCookie(req),
```

Comment:
Since you created a new lib to determine if a cookie should be secure, wouldn't it make sense to also have a helper function that receives the `res` object and handles the other parts of cookies (e.g., the age and `httpOnly`) to reduce redundancy? This should make it easier if in the future a different setting needs to be handled in all instances of cookies.

Reply:
IMO, no. `httpOnly` will never change, and all of these cookies have different max ages.

Reply:
Sounds good. That said, I'd prefer not to review it as I don't have access to a computer with Safari to test it.